test/cuda-mpi-cpp/base/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+cmake_minimum_required(VERSION 3.14)
+
+project(cuda-mpi-cpp LANGUAGES CXX)
+
+set(BLT_CXX_STD c++14 CACHE STRING "")
+set(ENABLE_CUDA On CACHE BOOL "")
+
+# Load BLT
+include(${BLT_SOURCE_DIR}/SetupBLT.cmake)
+
+# Add third-party library setup files to the project's installation
+# directory.
+blt_install_tpl_setups(DESTINATION lib/cmake/${PROJECT_NAME})
+
+# Add the base library with its single source file and header file.
+blt_add_library(
+  NAME cuda-mpi-cpp
+  SOURCES cuda-mpi-cpp.cu
+  HEADERS cuda-mpi-cpp.cuh
+  DEPENDS_ON cuda)
```

Comment:
Based on the directory name, I was expecting to see `mpi` as a dependency here. If it's only in the `downstream` dependency, should the latter use the `cuda-arch` results instead of a new kernel?

Reply:
That's a great point, yes mpi is only a downstream dependency.  Unfortunately the way the runner is set up this is a little tricky to do.  Each test assumes one unique base project and another downstream.  Ideally this wouldn't be a hard requirement, but I think that in this case it makes more sense to just create a more nontrivial base project.

Reply:
another option would be to just make the base directory a symlink, but I'm not sure that this would be portable to windows machines